### ISSUE TITLE
Model training takes a long time ？

### ISSUE THREAD

**Issue by tensorflowt** [signatrix/efficientdet#32]
Hello! First of all, thank you very much for your open source project, very good! 

Secondly, I am more concerned about the model training time. My hardware resource is a nvidia 2080ti graphics card with 11GB of memory. At present, I have trained the parameters according to your model: epoch=500; Batch_size = 8. So far it took me about 8 hours to iterate 1.5 epochs. Is this normal? Thank you very much!

**Comment by florisdesmedt** [comment]
I reach around 1hour 40 minutes per epoch (coco 2017 dataset) on nvidia rtx2070. So seems very slow on your system. Are images on local disk? Possible I/O delay?

**Comment by tensorflowt** [comment]
My system is ubuntu16.04, and my data is stored according to the author's project catalog. I want to ask, what is your model training parameter: batch_size?

**Comment by florisdesmedt** [comment]
I have experimented with different batch sizes, mostly the default 8.

**Comment by rmcavoy** [comment]
@tensorflowt If you checkout my issue #30, you can a decent speedup by modifying the loss evaluation to allocate several large arrays directly on the gpu rather than the current setup which allocates them on cpu and then moves them to the gpu. It only involves changing 3 lines in a very straightforward manner.

**Comment by tensorflowt** [comment]
> @tensorflowt If you checkout my issue #30, you can a decent speedup by modifying the loss evaluation to allocate several large arrays directly on the gpu rather than the current setup which allocates them on cpu and then moves them to the gpu. It only involves changing 3 lines in a very straightforward manner.

Thank you very much for your suggestions, but I'm not quite sure how I should modify my project file to achieve your optimization solution? Can you give more detailed operation procedures?

Thank you very much!
Best wishes

**Comment by rmcavoy** [comment]
The lines I mentioned in issue #30 (please read that issue if you haven't already) are all in the src/loss.py in the FocalLoss class (lines 63-65, lines 78-80, lines 90-92). Replace them with corresponding statements as directed in issue #30.
